test/faktory_worker/job/job_integration_test.exs

```diff
@@ -0,0 +1,40 @@
+defmodule FaktoryWorker.JobIntegrationTest do
+  use ExUnit.Case
+
+  alias FaktoryWorker.Job
+
+  describe "perform_async/3" do
+    test "should send a new job to faktory" do
+      faktory_name = :"Test_#{random_string()}"
+      start_supervised!({FaktoryWorker, name: faktory_name, pool: [size: 1]})
+
+      queue_name = random_string()
+
+      opts = [
+        queue: queue_name,
+        faktory_name: faktory_name
+      ]
+
+      job = Job.create_job(SomeWorker, %{hey: "there!"}, opts)
+
+      Job.perform_async(job, opts)
```

Comment:
Is it right that the same opts gets passed into the job create and into the perform async?

Reply:
Yes they currently just get the `opts` that get passed into `perform_async` on the worker that has `use FaktoryWorker.Job`. This is testing the function directly and not through the using approach (which I would like to test) which is why it looks a bit broken up.